syft/pkg/relations.go

```diff
@@ -0,0 +1,6 @@
+package pkg
+
+type Relations struct {
+	// ParentsByFileOwnership lists all parent packages that claim ownership of this package
```

Comment:
Reading this it seems that _Parents_ is needed because there can be more than one parent for a package, but that doesn't seem right for the type of relationship this is trying to establish (many-to-one). This is arguable, but I tend to think about a _Parent_ as a one way relationship to a _child_ or _children_ (package or packages in this case). 

Is it possible to rethink these variables to map the same concept?

Reply:
In a pure sense, what you are saying is true --there should really be a single parent and a possibility of multiple children. However, it is allowable for multiple packages to claim ownership of the same files, which would register here 
 as multiple parents; whether this is valid for any particular ecosystem to do doesn't necessarily matter.

I thought about making this something like:
```
type Relations struct {
   ChildrenByFileOwnership []ID
}
```
But this has the downside that consumers have to do a bit of work to determine which packages are "root" packages (not owned by another package). By tracking parent relationships a consumer only needs to check that the `ParentsByFileOwnership` is empty to ensure it is a "root" package.